kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/SessionMetadataDoc.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2024 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.internal.db.mongo;
+
+import java.util.Objects;
+
+/**
+ * The 'id' of a session metadata doc corresponds to the kafka partition,
+ * which is why we don't have to include the partition as a separate
+ * field for the metadata docs the way we do for the data docs
+ */
+public class SessionMetadataDoc {
```

Comment:
I think it might make sense to share this and `WindowMetadataDoc` for now - including all the code in `MongoSessionTable` related to metadata management

Reply:
I agree that this needs to be consolidated. I wrote this down and will address in a follow-up PR.